framework/include/st_things/st_things_types.h

```diff
@@ -0,0 +1,317 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#ifndef __ST_THINGS_TYPES_H__
+#define __ST_THINGS_TYPES_H__
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+/**
+ * @brief Enumeration for ST Things error code.
+ */
+typedef enum {
+	ST_THINGS_ERROR_NONE = 0,				  /**< Successful */
+	ST_THINGS_ERROR_INVALID_PARAMETER = -1,	  /**< Invalid parameter (If parameter is null or empty)*/
+	ST_THINGS_ERROR_OPERATION_FAILED = -2,	  /**< Operation Failed */
+} st_things_error_e;
+
+/**
+ * @brief Enumeration for ST Things status.
+ */
+typedef enum {
+	ST_THINGS_STATUS_INIT = 0,						   /**< Initial state of ST Things */
+	ST_THINGS_STATUS_ES_STARTED,					   /**< Easy-setup is started */
+	ST_THINGS_STATUS_ES_DONE,						   /**< Easy-setup is done */
+	ST_THINGS_STATUS_ES_FAILED_ON_OWNERSHIP_TRANSFER,  /**< Easy-setup failed due to Ownership-Transfer failure */
+	ST_THINGS_STATUS_CONNECTING_TO_AP,				   /**< Connecting to target Wi-Fi access point */
+	ST_THINGS_STATUS_CONNECTED_TO_AP,				   /**< Connected to target Wi-Fi access point */
+	ST_THINGS_STATUS_CONNECTING_TO_AP_FAILED,		   /**< Failed to connect to target Wi-Fi access point */
+	ST_THINGS_STATUS_REGISTERING_TO_CLOUD,			   /**< Trying to Sign-up/Sign-in/Publish-Resource(s) to Cloud */
+	ST_THINGS_STATUS_REGISTERED_TO_CLOUD,			   /**< Publish resource(s) to cloud is complete. Now the Thing is ready to be controlled via Cloud */
+	ST_THINGS_STATUS_REGISTERING_FAILED_ON_SIGN_IN,	   /**< Failed to sign-in to Cloud */
+	ST_THINGS_STATUS_REGISTERING_FAILED_ON_PUB_RES	   /**< Failed to publish resources to Cloud */
+	/**
+     * @todo More status needs to be added, especially for failure cases like 'incorrect password to connect to target AP'
+     */
+} st_things_status_e;
+
+/**
+ * @brief Structure for Representation.
+ */
+typedef struct _st_things_representation {
+	void *payload;	 /**< Payload of representation */
+
+	/**
+     * @brief API for getting the value of string type property with a key.
+     * @remarks This API will return deep-copied string value as out parameter, so application must free it after use.
+     * @param[in]  rep Instance of Representation.
+     * @param[in]  key Property Name which represents the value.
+     * @param[out] value String value
+     * @return @c true if value exist, otherwise @c false
+     */
+	 bool(*get_str_value) (struct _st_things_representation * rep, const char *key, char **value);
```

Comment:
1. add space between function type and funtion name.
2. remove space between function name and arguments
3. remove space between indicator of pointer and argument name (* rep -> *rep)
4. remove space at start of definition
```
 bool(*get_str_value) (struct _st_things_representation * rep, const char *key, char **value);
bool (*get_str_value)(struct _st_things_representation *rep, const char *key, char **value);
```